specification/hybridconnectivity/resource-manager/Microsoft.HybridConnectivity/stable/2023-03-15/hybridconnectivity.json

```diff
@@ -0,0 +1,1093 @@
+      "properties": {
+        "value": {
+          "description": "The list of service configuration",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ServiceConfigurationResource"
+          }
+        },
+        "nextLink": {
+          "description": "The link to fetch the next page of connected cluster",
+          "type": "string"
+        }
+      }
+    },
+    "EndpointResource": {
+      "description": "The endpoint for the target resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "systemData": {
+          "description": "System data of endpoint resource",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "x-ms-client-flatten": true
+        },
+        "properties": {
+          "description": "The endpoint properties.",
+          "$ref": "#/definitions/EndpointProperties",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ServiceConfigurationResource": {
+      "description": "The service configuration details associated with the target resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "systemData": {
+          "description": "System data of endpoint resource",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "x-ms-client-flatten": true
+        },
+        "properties": {
+          "description": "The service configuration properties.",
+          "$ref": "#/definitions/ServiceConfigurationProperties",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ServiceConfigurationProperties": {
+      "type": "object",
+      "description": "Service configuration details",
+      "required": [
+        "serviceName"
+      ],
+      "properties": {
+        "serviceName": {
+          "type": "string",
+          "description": "Name of the service.",
+          "enum": [
+            "SSH",
+            "WAC"
+          ],
+          "x-ms-enum": {
+            "name": "serviceName",
+            "modelAsString": true
+          }
+        },
+        "resourceId": {
+          "type": "string",
+          "description": "The resource Id of the connectivity endpoint (optional)."
+        },
+        "port": {
+          "type": "string",
+          "pattern": "^[0-9]+$",
+          "description": "The port on which service is enabled."
+        },
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The resource provisioning state.",
+          "enum": [
+            "Succeeded",
```

Comment:
>Succeeded [](http://example.com/codeflow?start=13&length=9)

Why do you only define terminal states, and no non-terminal state, like Pending, InProgress, Accepted, Creating, Deleting, etc.?

Reply:
Or just not have a provisioningState if it always provisions immediately.

Reply:
Added the non-terminal state that would be best as per our scneario.